sim/battle.ts

```diff
@@ -2954,8 +2966,28 @@ export class Battle extends Dex.ModdedDex {
 		}
 	}
 
+	// omniscient = -1, spectator = 0
+	static updateForSide(data: string, side: -1 | 0 | 1 | 2 | 3 | 4 = 0) {
```

Comment:
`extractChannel` seems to be the better name here.

This name is kind of misleading since I'd expect functions starting with `update` to be impure, and return a success state.

Reply:
AIUI, there aren't "channels" in the sim (unlike in the server), the closet analog is side. Would you be OK with `extractUpdateForSide`?

Reply:
Considering you're passing a channel ID to this function, this function introduces channels to the sim. The function should probably take a side ID if you didn't want to do that.